### ISSUE TITLE
SA1500 false positive in stackalloc expression

### ISSUE THREAD

**Issue by Tan90909090** [DotNetAnalyzers/StyleCopAnalyzers#3047]
Version used: `StyleCop.Analyzers 1.2.0-beta.66`
```cs
using System;

class C
{
    static void Main()
    {
        Foo(stackalloc[] { 'a' }); // SA1500 is reported here
        Foo(new[] { 'a' }); // SA1500 is NOT reported here
    }

    static void Foo(ReadOnlySpan<char> a) { }
}
```
The `Foo(stackalloc[] { 'a' });` code is in one line, so I think SA1500 should not be reported that.
